doc/versioned-translated.sh

```diff
@@ -0,0 +1,120 @@
+   export current_version
+
+   changed=false
+
+   git checkout ${current_version}
+  
+   echo "INFO: Building sites for ${current_version}"
+  
+   # skip this branch if it doesn't have our docs dir & sphinx config
+   if [ ! -e 'source/conf.py' ]; then
+      echo "INFO: Couldn't find 'doc/source/conf.py' (skipped)"
+      continue
+   fi
+  
+   for current_language in ${languages}; do
+  
+      # make the current language available to conf.py
+      export current_language
+  
+      echo "INFO: Building for ${current_language}"
+
+      if [[ ! -f "locales/$current_language/LC_MESSAGES/framework-docs.po" ]]; then
+        if [[ $current_language != "en" ]]; then
+          echo "No translation, using default one"
+          rm -rf locales/$current_language
+          mkdir -p locales
+          cp -r ${tmp_dir}/locales/$current_language locales/
+          echo "locale_dirs = ['../locales']" >> source/conf.py
+          echo "gettext_compact = 'framework-docs'" >> source/conf.py
+          make gettext
+          sphinx-intl update -p build/gettext -l ${current_language}
+        fi
+        cp -r ${tmp_dir}/_templates source
+
+        echo "html_context = dict()" >> source/conf.py
+        echo "html_context['current_language'] = '$current_language'" >> source/conf.py
+        echo "from git import Repo" >> source/conf.py
+        echo "repo = Repo( search_parent_directories=True )" >> source/conf.py
+        echo "current_version = '$current_version'" >> source/conf.py
+        echo "html_context['current_version'] = {}" >> source/conf.py
+        echo "html_context['current_version']['name'] = 'Main' if current_version=='main' else current_version" >> source/conf.py
+        echo "html_context['current_version']['url'] = current_version" >> source/conf.py
+        echo "html_context['current_version']['full_name'] = 'Main' if current_version=='main' else f'Flower Framework {current_version}'" >> source/conf.py
+        echo "html_context['versions'] = list()" >> source/conf.py
+        echo "versions = [tag.name for tag in repo.tags if int(tag.name[1]) != 0]" >> source/conf.py
+        echo "versions.append('main')" >> source/conf.py
+        echo "for version in versions:" >> source/conf.py
+        echo "    html_context['versions'].append({'name': version})" >> source/conf.py
+        echo "html_sidebars = {" >> source/conf.py
+        echo "    '**': [" >> source/conf.py
+        echo "        'sidebar/brand.html'," >> source/conf.py
+        echo "        'sidebar/search.html'," >> source/conf.py
+        echo "        'sidebar/scroll-start.html'," >> source/conf.py
+        echo "        'sidebar/navigation.html'," >> source/conf.py
+        echo "        'sidebar/scroll-end.html'," >> source/conf.py
+        echo "        'sidebar/versioning.html'," >> source/conf.py
+        echo "        'sidebar/lang.html'," >> source/conf.py
+        echo "    ]" >> source/conf.py
+        echo "}" >> source/conf.py
```

Comment:
```suggestion
tee -a source/conf.py << END
        html_context = dict()
        html_context['current_language'] = '$current_language'
        from git import Repo
        repo = Repo( search_parent_directories=True )
        current_version = '$current_version'
        html_context['current_version'] = {}
        html_context['current_version']['name'] = 'Main' if current_version=='main' else current_version
        html_context['current_version']['url'] = current_version
        html_context['current_version']['full_name'] = 'Main' if current_version=='main' else f'Flower Framework {current_version}'
        html_context['versions'] = list()
        versions = [tag.name for tag in repo.tags if int(tag.name[1]) != 0]
        versions.append('main')
        for version in versions:
            html_context['versions'].append({'name': version})
        html_sidebars = {
            '**': [
                'sidebar/brand.html',
                'sidebar/search.html',
                'sidebar/scroll-start.html',
                'sidebar/navigation.html',
                'sidebar/scroll-end.html',
                'sidebar/versioning.html',
                'sidebar/lang.html',
            ]
        }
END
```

Reply:
Not 100% sure if I missed something but this might be easier to read / maintain.